src/api/update.cpp

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2019, QuantStack and Mamba Contributors
+//
+// Distributed under the terms of the BSD 3-Clause License.
+//
+// The full license is in the file LICENSE, distributed with this software.
+
+#include "mamba/api/install.hpp"
+
+#include "mamba/core/context.hpp"
+#include "mamba/api/configuration.hpp"
+#include "mamba/core/virtual_packages.hpp"
+
+
+namespace mamba
+{
+    void update(bool update_all)
+    {
+        auto& ctx = Context::instance();
+        auto& config = Configuration::instance();
+
+        config.load(MAMBA_ALLOW_ROOT_PREFIX | MAMBA_ALLOW_FALLBACK_PREFIX
+                    | MAMBA_ALLOW_EXISTING_PREFIX);
+
+        auto update_specs = config.at("specs").value<std::vector<std::string>>();
+
+        if (update_all)
+        {
+            PrefixData prefix_data(ctx.target_prefix);
+            prefix_data.load();
+
+            for (const auto& package : prefix_data.m_package_records)
+            {
+                auto name = package.second.name;
+                if (name != "python")
```

Comment:
this is probably not new, but I just wanted to note that we shouldn't just remove python here.

It shoudl work to update python minor versions with the python-pinning behavior you implemented, right?

Reply:
> I just wanted to note that we shouldn't just remove python here.

I don't get it

This is done to not update `python` when using `-a,--all`/`update_all`

Reply:
I think we do want to update python, but keep the minor version. 

So python 3.7.4 should update to 3.7.8, but not to 3.9.1

Is that a wrong assumption?